lib/osut/utils.rb

```diff
@@ -5887,50 +6071,180 @@ def getHorizontalRidges(roofs = [])
     ridges
   end
 
+  ##
+  # Preselects ideal spaces to toplight, based on 'addSkylights' options and key
+  # building model geometry attributes. Can be called from within 'addSkylights'
+  # by setting :ration (opts key:value argument) to 'true' ('false' by default).
+  # Alternatively, the method can be called prior to 'addSkylights'. The set of
+  # filters stems from previous rounds of 'addSkylights' stress testing. It is
+  # intended as an option to prune away less ideal candidate spaces (irregular,
+  # smaller) in favour of (larger) candidates (notably with more suitable
+  # roof geometries). This is key when dealing with attic and plenums, where
+  # 'addSkylights' seeks to add skylight wells (relying on roof cut-outs and
+  # leader lines). Another check/outcome is whether to prioritize skylight
+  # allocation in already sidelit spaces - opts[:sidelit] may be reset to 'true'.
+  #
+  # @param spaces [Array<OpenStudio::Model::Space>] candidate(s) to toplight
+  # @param [Hash] opts requested skylight attributes (same as 'addSkylights')
+  # @option opts [#to_f] :size (1.22m) template skylight width/depth (min 0.4m)
+  #
+  # @return [Array<OpenStudio::Model::Space>] candidates (see logs if empty)
+  def toToplit(spaces = [], opts = {})
```

Comment:
An option to _addSkylights_, yet called upon by default. This is useful when treating multiple 3rd-party models (e.g. a specific market segment), with undiagnosed geometry (i.e. flying blind). Through trial and error, these handful of sliders and rules have yielded satisfactory results during recent stress testing.